File: diagonal.go
package main

import "fmt"

func diagonalDifference(arr [][]int32) int32 {
	// Write your code here
	a := int32(0)
	b := int32(0)
	index := 0
	length := 0
	for _, each_arr := range arr {
		length = len(each_arr)
		//fmt.Println(each_arr)
		a += each_arr[index]
		//fmt.Println(each_arr[index])
		index += 1
	}
	length -= 1
	for _, each_arr := range arr {
		//fmt.Println(each_arr)
		b += each_arr[length]
		//fmt.Println(each_arr[length])
		length -= 1
	}
	num := a - b
	if num < 0 {
		num *= -1
	}
	return num
}

func main() {
	fmt.Println(diagonalDifference([][]int32{{11, 2, 4}, {4, 5, 6}, {10, 8, -12}}))
}

File: ALGORITHMS/diagonal.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'diagonalDifference' function below.
#
# The function is expected to return an INTEGER.
# The function accepts 2D_INTEGER_ARRAY arr as parameter.
#

def diagonalDifference(arr):
    # Write your code here
    left = 0
    right = 0
    idx = 0
    for each_arr in arr:
        index = len(each_arr)-1
        print(each_arr[idx])
        left += each_arr[idx]
        idx += 1
    for each_arr in arr:
        right += each_arr[index]
        print(each_arr[index])
        index -= 1

    num = left - right
    if num < 0:
        num *= -1
        
    return num
        
        
        

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    arr = []

    for _ in range(n):
        arr.append(list(map(int, input().rstrip().split())))

    result = diagonalDifference(arr)

    fptr.write(str(result) + '\n')

    fptr.close()
